client/android/WorldScope/app/src/main/java/com/litmus/worldscope/RecordActivity.java

```diff
@@ -0,0 +1,470 @@
+            prev_rw = display_width_d;
+            prev_rh = (int) (1.0 * display_width_d * live_height / live_width);
+        }
+        frameLayoutParam = new FrameLayout.LayoutParams(prev_rw, prev_rh);
+        frameLayoutParam.topMargin = (int) (1.0 * bg_screen_by * screenHeight / bg_height);
+        frameLayoutParam.leftMargin = (int) (1.0 * bg_screen_bx * screenWidth / bg_width);
+
+        //Set the camera to portrait mode
+        cameraDevice = Camera.open();
+        cameraDevice.setDisplayOrientation(ROTATION_90);
+
+        Log.i(LOG_TAG, "Camera open");
+        cameraView = new CameraView(this, cameraDevice);
+        topLayout.addView(cameraView, frame LayoutParam);
+        Log.i(LOG_TAG, "Camera preview start: OK");
+
+        // TODO: Tweak start buttons
+        //Add in the button bar into the FrameLayout
+        relativeLayoutParam = new RelativeLayout.LayoutParams(screenWidth, screenHeight);
+        RelativeLayout streamButtonBar = (RelativeLayout) myInflate.inflate(R.layout.template_stream_button_bar, null);
+        topLayout.addView(streamButtonBar, relativeLayoutParam);
+    }
+
+    // Start recording
+    private void startRecorder() {
+        initializeRecorder();
+        try {
+            recording = true;
+            recorder.start();
+            startTime = System.currentTimeMillis();
+            audioThread.start();
+
+        } catch (FFmpegFrameRecorder.Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    // Stop recording
+    public void stopRecording() {
+
+        runAudioThread = false;
+        try {
+            audioThread.join();
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+        audioRecordRunnable = null;
+        audioThread = null;
+
+        if (recorder != null && recording) {
+
+            recording = false;
+            Log.v(LOG_TAG,"Finishing recording, calling stop and release on recorder");
+            try {
+                recorder.stop();
+                recorder.release();
+            } catch (FFmpegFrameRecorder.Exception e) {
+                e.printStackTrace();
+            }
+            recorder = null;
+
+        }
+    }
+
+    // Create recorder
+    private void initializeRecorder() {
+        Log.w(LOG_TAG,"init recorder");
+
+        if (yuvImage == null) {
+            yuvImage = new Frame(imageWidth, imageHeight, Frame.DEPTH_UBYTE, 2);
+            Log.i(LOG_TAG, "create yuvImage");
+        }
+
+        Log.i(LOG_TAG, "ffmpeg_url: " + ffmpeg_link);
+        recorder = new FFmpegFrameRecorder(ffmpeg_link, imageWidth, imageHeight, 1);
+
+        // Custom format
+        recorder.setFormat("flv");
+        recorder.setVideoCodec(28);
+        recorder.setAudioCodec(86018);
+        recorder.setSampleRate(22050);
+        recorder.setFrameRate(30.0D);
+
+        // Default format
+        //recorder.setSampleRate(sampleAudioRateInHz);
+        //recorder.setFrameRate(frameRate);
```

Comment:
These could be declared as constants, so it will be easier for you to refer to or change later on